eng/pipelines/templates/steps/analyze.yml

```diff
@@ -16,7 +16,7 @@ steps:
       pwsh: true
       workingDirectory: $(Pipeline.Workspace)
     displayName: Dump Package properties
-    condition: succeeded()
+    condition: and(succeeded(), not(contains('${{ parameters.ServiceDirectory }}', '/eng/')))
```

Comment:
I don't know how this fails but perhaps we just make sure our script can handle service directories that don't have any packages.

Reply:
It fails at https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Save-Package-Properties.ps1#L124 when it returns empty package properties. So we could move the exclusion for the `eng` directory into the script itself, but it seems like it's intended behavior that it would fail in other situations vs. passing silently. I assume in most/all places where this script is used there are downstream steps that rely on the package properties not being empty.

It would be cleaner in the script from the go pipeline configs, but at the same time it is putting an exception for a single language into a core component, so ¯\_(ツ)_/¯ Thoughts?

Reply:
That is fair. I wouldn't add an "eng" special casing inside of the script I was just wondering if there are assumptions built in that would break if we just let an empty list come back. 

Do you like adding a "/eng/" explicit condition here or just a "Skip" variable that we could set in those eng pipelines?

Reply:
I think Skip is probably cleaner, but we also don't have a config driven way to set that up, so it could make new eng tool pipelines break in the future. I'm not the biggest fan of this current `/eng/` skip condition, so maybe it's worth the trade-off?

Reply:
I was thinking of some skip variable would make this more opt-in to the exception cases which I like better then adding this special cases. If things under /eng/ is special enough maybe we create a template with this variable setup for those pipelines.

Reply:
@weshaggard I added a separate pipeline. I think that will be the most flexible for handling similar situations that come up in the future.